src/lib/Libutil/hook.c

```diff
@@ -1456,19 +1472,26 @@ add_hook_event(hook *phook, char *newval, char *msg, size_t msg_len)
 			phook->event 	|= HOOK_EVENT_EXECJOB_RESIZE;
 			insert_hook_sort_order(HOOK_EVENT_EXECJOB_RESIZE,
 				&svr_execjob_resize_hooks, phook);
+		} else if (strcmp(val, HOOKSTR_EXECJOB_ABORT) == 0) {
+			if (phook->event & HOOK_EVENT_PROVISION)
+				goto err;
+			delete_link(&phook->hi_execjob_abort_hooks);
+			phook->event 	|= HOOK_EVENT_EXECJOB_ABORT;
+			insert_hook_sort_order(HOOK_EVENT_EXECJOB_ABORT,
+				&svr_execjob_abort_hooks, phook);
```

Comment:
This part can also be refactored using a loop. Only provision hook needs to be handled differently.

Reply:
@minghui-liu:  I will file an issue to refactor code in hook.c to be worked on later, separate from this PR so as to not broaden the scope of this and introduce larger risk.

Reply:
Filed PP-1334: Refactor Libutil/hook.c code to eliminate repetitive if-else blocks.